src/ui/public/vislib/lib/axis.js

```diff
@@ -68,7 +70,7 @@ export default function AxisFactory(Private) {
     }
 
     getLength(el, n) {
-      const margin = this.config.get('vis._attr.margin');
+      const margin = this.config._chartConfig.get('style.margin');
```

Comment:
in this specific case, we are not talking about axis margins, we talk about overall vis margin, but axis needs to know about it as well in order to render correctly. i agree accessing chartconfig thru axisconfig doesnt make much sense, its just a quick way i replaced old code (accessing vis._attr directly) with something that works. i plan to extract those out.

so in my opinion there should be a VisConfig .... to which pretty much every component needs access ... and axisConfig should be like a subset of this configuration but with its own defaults/helper methods ... which are only needed by axis.